design-proposals/default-instancetypes-from-volumes.md

```diff
@@ -0,0 +1,89 @@
+# Overview
+
+[Instancetypes and preferences](https://kubevirt.io/user-guide/virtual_machines/instancetypes/) allow a users to pick a set of predefined resource, performance and other runtime preferences for their `VirtualMachine` without having to worry about the complexity of KubeVirt's API.
+
+This design document aims to outline a simple way this feature could be further extended by allowing users to opt into having a default instancetype and/or preference inferred from a given `Volume` associated with their `VirtualMachine`.
+
+## Motivation
+
+The inference of a default instancetype and preference from a `Volume` associated with the `VirtualMachine` further reduces the choices a user has to make in order to get a running `VirtualMachineInstance` to a singular choice about which runnable image they would like to boot.
+
+## Goals
+
+* Allow users to opt into having a default instancetype and preference inferred from a given `Volume` associated with their `VirtualMachine`.
+
+## Non Goals
+
+* Automatic annotation of a `Volume` or underlying type (such as a `PVC`) is not covered as part of this work but is something that will be investigated in the future.
+
+## User Stories
+
+* As an admin, user or third party creator of a `Volume`,  I want to provide annotations that suggest the default instancetype or set of preferences a `VirtualMachine` should use when booting from the `Volume`.
+
+* As a user, I want to opt into allowing KubeVirt to infer these defaults from a specific volume associated with my VirtualMachine.
+
+## Repos
+
+* [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+The `InstancetypeMatcher` and `PreferenceMatcher` types will be extended to include a new `inferFromVolume` attribute that will control this behaviour.
+
+When provided the `Volume` referred to by this attribute will be checked for the following annotations:
+
+* `instancetype.kubevirt.io/defaultInstancetype`
+* `instancetype.kubevirt.io/defaultInstancetypeKind` (Defaults to `VirtualMachineClusterInstancetype`)
+* `instancetype.kubevirt.io/defaultPreference`
+* `instancetype.kubevirt.io/defaultPreferenceKind` (Defaults to `VirtualMachineClusterPreference`)
```

Comment:
what happens if these annotations change on the PVC?  is inferFromVolume  a one-shot deal? or do we keep inferring after vm creation?

Reply:
As discussed with @dominikholler we can drop the `inferFromVolume` from the matcher with the initial lookup and application in the mutation webhook.